gateway-ha/src/main/java/io/trino/gateway/ha/router/RuleReloadingRoutingGroupSelector.java

```diff
@@ -84,9 +87,13 @@ public String findRoutingGroup(HttpServletRequest request)
                     writeLock.unlock();
                 }
             }
+            ProcessedRequest processedRequest = new ProcessedRequest(request, processedRequestConfig);
```

Comment:
The body processing is restricted to [POST](https://github.com/trinodb/trino-gateway/pull/325/commits/c100ecdf150c6cad7a465765699d3016d4a2173e#diff-973162668135a5cdba92480352158a4562d54c8815725d9fc0f852703202f4d8R297). The `processedRequest` object needs to be passed to the rules engine everytime it is invoked, because a rule referencing a nonexistent object will produce a runtime error. However, I could add a method like `boolean isPost()` that could be checked in a rule - would that be helpful?